service/awsconfig/v3/resource_set.go

```diff
@@ -0,0 +1,309 @@
+	var err error
+
+	if config.CertsSearcher == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.CertsSearcher must not be empty")
+	}
+	if config.GuestAWSClients.CloudFormation == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.GuestAWSClients.CloudFormation must not be empty")
+	}
+	if config.GuestAWSClients.EC2 == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.GuestAWSClients.EC2 must not be empty")
+	}
+	if config.GuestAWSClients.ELB == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.GuestAWSClients.ELB must not be empty")
+	}
+	if config.GuestAWSClients.IAM == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.GuestAWSClients.IAM must not be empty")
+	}
+	if config.GuestAWSClients.KMS == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.GuestAWSClients.KMS must not be empty")
+	}
+	if config.GuestAWSClients.S3 == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.GuestAWSClients.S3 must not be empty")
+	}
+	if config.HostAWSClients.CloudFormation == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.HostAWSClients.CloudFormation must not be empty")
+	}
+	if config.HostAWSClients.EC2 == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.HostAWSClients.EC2 must not be empty")
+	}
+	if config.HostAWSClients.IAM == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.HostAWSClients.IAM must not be empty")
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.RandomkeysSearcher == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.RandomkeysSearcher must not be empty")
+	}
+
+	if len(config.HandledVersionBundles) == 0 {
+		return nil, microerror.Maskf(invalidConfigError, "config.HandledVersionBundles must not be empty")
+	}
+	if config.InstallationName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.InstallationName must not be empty")
+	}
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.ProjectName must not be empty")
+	}
+
+	var awsService *awsservice.Service
+	{
+		c := awsservice.Config{
+			Clients: awsservice.Clients{
+				IAM: config.GuestAWSClients.IAM,
+				KMS: config.GuestAWSClients.KMS,
+			},
+			Logger: config.Logger,
+		}
+
+		awsService, err = awsservice.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var cloudConfig *cloudconfig.CloudConfig
+	{
+		c := cloudconfig.Config{
+			Logger: config.Logger,
+		}
+
+		cloudConfig, err = cloudconfig.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var kmsKeyResource framework.Resource
+	{
+		c := kmskey.Config{
+			Clients: kmskey.Clients{
+				KMS: config.GuestAWSClients.KMS,
```

Comment:
And here we see why this is just better. 

```
c := Config{}

c.Foo.Bar.Baz = fooBarBaz
...
```

Reply:
I wouldn't call that better :) I like to see alignment and structure, but that's personal.
This is just how you do it in Go, people are used to that and I don't think adding some personal touch is beneficial. As I said when team says this is the way to go, I follow.

Reply:
I lost the battle. I will comply with the team from now on. 😢